NppExec/src/DlgAdvancedOptions.cpp

```diff
@@ -391,96 +378,41 @@ void CAdvOptDlg::OnInitDlg(HWND hDlg)
     advOptDlg.CenterWindow(NppExec.m_nppData._nppHandle);
 }
 
+template<class OM, class W> static void SaveColorOption(

```

Comment:
Do we really need a templated function? Probably it could be a regular (non-templated) function.

Reply:
It does not **HAVE** to be a template but it was less typing and it is more generic, it does not care that the options manager is Static (whatever that means in this context), all it cares about is that it has a SetData method.

Reply:
I'd prefer non-template implementation. The CStaticOptionsManager instance is returned by CNppExec::GetOptions() and it is the only options manager used by NppExec. The word "static" is related to static array of options handled by CStaticOptionsManager, i.e. it does not allow to add new options at runtime or remove existing options at runtime.